java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java

```diff
@@ -1029,4 +1031,164 @@ CompletableFuture<ClusterValue<Object>> fcallReadOnly(
+     *     href="https://valkey.io/commands/evalsha/">EVALSHA</a> for details.
+     * @param script The Lua script to execute.
+     * @param args The non-key arguments for the script.
+     * @param route Specifies the routing configuration for the command. The client will route the
+     *     command to the nodes defined by <code>route</code>.
+     * @return a value that depends on the script that was executed.
+     * @example
+     *     <pre>{@code
+     * try(Script luaScript = new Script("return { ARGV[1] }", false)) {
+     *     Object[] result = (Object[]) client.invokeScript(luaScript, {"foo"}, ALL_PRIMARIES).get();
+     *     assert result[0].equals("foo");
+     * }
+     * }</pre>
+     */
+    CompletableFuture<Object> invokeScript(Script script, List<String> args, Route route);
+
+    /**
+     * Invokes a Lua script with its keys and arguments.<br>
+     * This method simplifies the process of invoking scripts on the server by using an object that
+     * represents a Lua script. The script loading, argument preparation, and execution will all be
+     * handled internally. If the script has not already been loaded, it will be loaded automatically
+     * using the <code>SCRIPT LOAD</code> command. After that, it will be invoked using the <code>
+     * EVALSHA</code> command.
+     *
+     * @apiNote When in cluster mode
+     *     <ul>
+     *       <li>all <code>keys</code> must map to the same hash slot.
+     *       <li>if no <code>keys</code> are given, command will be routed to a random primary node.
+     *     </ul>
+     *
+     * @see <a href="https://valkey.io/commands/script-load/">SCRIPT LOAD</a> and <a
+     *     href="https://valkey.io/commands/evalsha/">EVALSHA</a> for details.
+     * @param script The Lua script to execute.
+     * @param args The non-key arguments for the script.
+     * @param route Specifies the routing configuration for the command. The client will route the
+     *     command to the nodes defined by <code>route</code>.
+     * @return a value that depends on the script that was executed.
+     * @example
+     *     <pre>{@code
+     * try(Script luaScript = new Script(gs("return { ARGV[1] }", true))) {
+     *     Object[] result = (Object[]) client.invokeScript(luaScript, {"foo"}, ALL_PRIMARIES).get();
+     *     assert result[0].equals(gs("foo"));
+     * }
+     * }</pre>
+     */
+    CompletableFuture<Object> invokeScriptBinary(Script script, List<GlideString> args, Route route);
+
+    /**
+     * Check existence of scripts in the script cache by their SHA1 digest.
+     *
+     * @see <a href="https://valkey.io/commands/script-exists">SCRIPT EXISTS</a> for details.
+     * @param sha1s The Lua script to execute.
+     * @param route The command will be routed to all primary nodes, unless <code>route</code> is
+     *     provided, in which case the client will route the command to the nodes defined by <code>
+     *     route</code>. Defaults to None.
```

Comment:
Copy route doc from other java commands. There is no option to not specifying a route for this API.

Reply:
You need to add `scriptExists` without route argument and with extra doc line about default routing.
Applicable to all commands for cluster client

Reply:
Update this now

Reply:
its above this api